src/pykiso/test_result/assert_step_report.py

```diff
@@ -232,6 +233,35 @@ def _add_step(
     )
 
 
+def determine_parent_test_function(test_name: str) -> str:
```

Comment:
Can you provide some more information about this function? More towards time-complexity, how fast are we or how slow did we get?

Reply:
how fast are we or how slow did we get? In the case of nested assert we get slower (more code to execute). Otherwise (assert method not nested) nothing really significant.

Based on the function and the call of the stack a few ms (10 -20 ms) more if the assert method is nested.

Otherwise (assert method not nested) a just the if comparison so a few micro seconds.